engine/src/main/java/com/datatorrent/stram/util/WebServicesClient.java

```diff
@@ -61,34 +66,62 @@
 
   private static final PoolingHttpClientConnectionManager connectionManager;
   private static final CredentialsProvider credentialsProvider;
+  private static final RegistryBuilder<AuthSchemeProvider> registryBuilder;
+  private static Lookup<AuthSchemeProvider> authRegistry;
   private static final int DEFAULT_CONNECT_TIMEOUT = 10000;
   private static final int DEFAULT_READ_TIMEOUT = 10000;
 
   private final Client client;
 
   private final Set<ClientFilter> clientFilters = new HashSet<>();
 
+  private static final Credentials DEFAULT_TOKEN_CREDENTIALS = new Credentials()
+  {
+    @Override
+    public Principal getUserPrincipal()
+    {
+      return null;
+    }
+
+    @Override
+    public String getPassword()
+    {
+      return null;
+    }
+  };
+
   static {
     connectionManager = new PoolingHttpClientConnectionManager();
     connectionManager.setMaxTotal(200);
     connectionManager.setDefaultMaxPerRoute(5);
+    registryBuilder = RegistryBuilder.<AuthSchemeProvider>create();
     credentialsProvider = new BasicCredentialsProvider();
-    credentialsProvider.setCredentials(AuthScope.ANY, new Credentials()
-    {
+    // By default add SPNEGO so that it works even if auth is not explictly configured like before, in future
+    // move it to auth setup below
+    registryBuilder.register(AuthSchemes.SPNEGO, new SPNegoSchemeFactory(true));
+    credentialsProvider.setCredentials(AuthScope.ANY, DEFAULT_TOKEN_CREDENTIALS);
+    authRegistry = registryBuilder.build();
+  }
 
-      @Override
-      public Principal getUserPrincipal()
-      {
-        return null;
-      }
+  public static void initAuth(ConfigProvider configuration)
+  {
+    // Adding BASIC auth
+    AuthScheme scheme = AuthScheme.BASIC;
+    String username = configuration.getProperty(scheme, "username");
+    String password = configuration.getProperty(scheme, "password");
+    if (username != null) {
```

Comment:
squashed.. there is explicit handling for null password in the http client library but not for username which leads me to belive password can be null not username. We may not want to log a warn message that password is missing for example because that can expose security related information which someone can misuse.